db.go

```diff
@@ -60,13 +60,42 @@ func Create(path string) (*DB, error) {
 // Open opens the database at the location path using mode. Caller is responsible
 // for closing the database when done.
 func Open(path string, mode DBMode) (*DB, error) {
-	cpath := C.CString(path)
-	defer C.free(unsafe.Pointer(cpath))
+	config := ""
+	return OpenWithConfig(&path, &config, mode)
+}
+
+// OpenWithConfig opens the database at the location path using mode using the
+// configuration in config.
+//
+// If path is nil, use the location specified:
+//	- in the environment variable NOTMUCH_DATABASE, if non-empty
+//  - in a configuration file, located as described in 'config'
+//  - by $XDG_DATA_HOME/notmuch/$PROFILE
+//
+// If config is nil, it will use the path:
+//	- in environment variable NOTMUCH_CONFIG, if non-empty
+//	- by $XDG_CONFIG_HOME/notmuch/
+//
+// If config is an empty string (""), then it will not open any configuration file.
+//
+// Caller is responsible for closing the database when done.
+func OpenWithConfig(path, config *string, mode DBMode) (*DB, error) {
+	var cpath *C.char
+	if path != nil {
+		cpath = C.CString(*path)
+		defer C.free(unsafe.Pointer(cpath))
+	}
+
+	var cconfig *C.char
+	if config != nil {
+		cconfig = C.CString(*config)
+		defer C.free(unsafe.Pointer(cconfig))
+	}
 
 	cmode := C.notmuch_database_mode_t(mode)
 	var cdb *C.notmuch_database_t
 	cdbptr := (**C.notmuch_database_t)(&cdb)
-	err := statusErr(C.notmuch_database_open(cpath, cmode, cdbptr))
+	err := statusErr(C.notmuch_database_open_with_config(cpath, cmode, cconfig, nil, cdbptr, nil))
```

Comment:
You are ignoring the error output var, what's your reasoning there?
While I understand that this works, it would be much nicer to add the error message from notmuch, meant for human consumption, to it and let the user know the details rather than just "doesn't work"